javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,204 @@
+];
+
+const reply = [
+    //0 
+    ["Hello!", "Hi!", "Hey!", "Hi there!"],
+    //1
+    [
+        "Fine... how are you?",
+        "Pretty well, how are you?",
+        "Fantastic, how are you?"
+    ],
+    //2
+    [
+        "Nothing much",
+        "Exciting things!"
+    ],
+    //3
+    ["Glad to hear it"],
+    //4
+    ["Why?", "Cheer up buddy"],
+    //5
+    ["What about?", "Once upon a time..."],
+    //6
+    ["You're welcome", "No problem"],
+    //7
+    ["Goodbye", "See you later"],
+    //8
+    ["I am a voice assistant", "A voice assistant"],
+];
+
+const alternative = [
+    "Same",
+    "Go on...",
+    "Try again",
+    "I'm listening...",
+    "Bro..."
+];
+const robot = ["How do you do, fellow human", "I am not a bot"];
+// EXTRA FEATURE ENDS HERE
+
+function getReply(command) {
+    let newCommand = command.toLowerCase();
+    let commandArray = newCommand.split(" ");
+
+    if (newCommand.includes("hello my name is ")) {
+        if (userName === commandArray[commandArray.length - 1]) {
+            return `I know you already ${userName}`
+        } else {
+            userName = commandArray[commandArray.length - 1];
+            return `Nice to meet you ${userName}`;
+        }
+    }
+
+    else if (newCommand === "what is my name") {
+        if (!userName) {
+            return `I don't know you.`;
+        } else {
+            return `Your name is ${userName}`;
+        }
+    }
+
+    else if (newCommand.startsWith("add") && newCommand.includes("todo")) {
+        todo = commandArray.slice(1, commandArray.indexOf("to"));
+        todo = todo.join(' ');
+        todoList.push(todo);
+        return `${todo} added to your todo`;
+    }
+
+    else if (newCommand.startsWith("remove") && newCommand.includes("todo")) {
+        todo = commandArray.slice(1, commandArray.indexOf("from"));
+        todo = todo.join(' ');
+        todoList.splice(todoList.indexOf(todo), 1);
+        return `removed ${todo} from your todo`;
+    }
+
+    else if (newCommand === "what is on my todo?") {
+        return `you have ${todoList.length} todos - ${todoList.join(" and ")}`;
+    }
+
+    else if (newCommand === "what day is it today?") {
+        return `${today.getDate()}. of ${month} ${today.getFullYear()}`;
+    }
+
+    else if (newCommand.startsWith("what is") && newCommand.includes("+") || newCommand.includes("-") || newCommand.includes("*") || newCommand.includes("/")) {
+        let calcArray = commandArray.slice(commandArray.indexOf("is") + 1);
+        const firstNum = parseInt(calcArray[0]);
+        const secondNum = parseInt(calcArray[2]);
+        const operator = calcArray[1];
+
+        if (operator == '+') {
+            return firstNum + secondNum;
+        }
+        else if (operator == '-') {
+            return firstNum - secondNum;
+        }
+        else if (operator == '*') {
+            return firstNum * secondNum;
+        }
+        else {
+            return firstNum / secondNum;
```

Comment:
If I type in `What is 4 $ 4?` then this default operation will output `4 / 4`. Can you add a default error handling else statement?